py/test/selenium/webdriver/common/alerts_tests.py

```diff
@@ -185,9 +185,16 @@ def test_should_allow_users_to_accept_an_alert_in_anested_frame(driver, pages):
     assert "Testing Alerts" == driver.title
 
 
-def test_should_throw_an_exception_if_an_alert_has_not_been_dealt_with_and_dismiss_the_alert():
-    pass
-    # //TODO(David) Complete this test
+@pytest.mark.xfail_safari
+def test_should_throw_an_exception_if_an_alert_has_not_been_dealt_with_and_dismiss_the_alert(driver, pages):
+    pages.load("alerts.html")
+    driver.find_element(By.ID, "alert").click()
+    WebDriverWait(driver, 5).until(EC.alert_is_present())
+
+    with pytest.raises(UnexpectedAlertPresentException):
+        driver.find_element(By.ID, "select").click()
+    alert = _wait_for_alert(driver)
```

Comment:
I'm confused about this line and the following. The click is supposed to automatically dismiss the alert by default, so there is no alert to wait for.

Reply:
I thought to do it explicitly due to the test name.

Reply:
Test that the alert has been dismissed

Reply:
Should be done now